zerver/views/documentation.py

```diff
@@ -105,22 +105,36 @@ def get_path(self, article: str) -> DocumentationArticle:
                 endpoint_method=None,
             )
 
-        if self.path_template == "/zerver/api/%s.md":
-            # Hack: `self.path_template` has a leading `/`, so we use + to add directories.
-            api_documentation_path = os.path.join(settings.DEPLOY_ROOT, "templates") + path
-            if not os.path.exists(api_documentation_path):
-                try:
-                    endpoint_name, endpoint_method = get_endpoint_from_operationid(article)
-                    path = "/zerver/api/api-doc-template.md"
-                except AssertionError:
-                    return DocumentationArticle(
-                        article_path=self.path_template % ("missing",),
-                        article_http_status=404,
-                        endpoint_path=None,
-                        endpoint_method=None,
-                    )
+        if os.path.exists(path):
```

Comment:
Alternatively, could this if/else block have an `and self.api_view` or something both here and in the next `elif`?

Reply:
I added the boolean `self.api_doc_view` to use for these checks, but also reworked the flow of this function overall in the updated version.